workloads/router-perf/csv_gen.py

```diff
@@ -0,0 +1,176 @@
+#!/usr/bin/python3
+import yaml
+from collections import defaultdict
+import csv
+import argparse
+import datetime
+import os
+import re
+from oauth2client.service_account import ServiceAccountCredentials
+import gspread
+from gspread_formatting import *
+
+regexp = re.compile(r"^\D*(\d+)\D*$")
+
+# Globals
+main_metric_rps = {"http": "avg(requests_per_second)", "edge": "avg(requests_per_second)", "mix": "avg(requests_per_second)"}
+label_map = {"http": "http", "edge": "edge", "mix": "mix"}
+main_metric_latency = {"http": "avg(latency_95pctl)", "edge": "avg(latency_95pctl)", "mix": "avg(latency_95pctl)"}
+
+def get_uuid_mapping(file_name):
+    uuid_map = {}
+    uuid_titles = []
+    with open(file_name, "r") as f:
+        for line in f.readlines():
+            if not uuid_titles:
+                uuid_titles = [title.strip() for title in line.split(",")]
+                continue
+            uuids = line.split(",")
+            for i, title in enumerate(uuid_titles):
+                uuid_map[uuids[i].strip()] = title
+    return uuid_map, uuid_titles
+
+
+def read_yaml(yaml_files):
+    data = {}
+    for yaml_file in yaml_files:
+        with open(yaml_file) as f:
+            match = regexp.match(yaml_file)
+            data[match.group(1) if match else yaml_file] = yaml.load(f, Loader=yaml.FullLoader)
+    return data
+
+
+def create_table_mappings(data, uuid_map, metric_map):
+    tables = {} 
+    for file_num, datum in data.items():
+        for test_type in datum["test_type"]:
+            for routes in datum["test_type"][test_type]["routes"]:
+                for cpt in datum["test_type"][test_type]["routes"][routes]["conn_per_targetroute"]:
+                    for keepalive in datum["test_type"][test_type]["routes"][routes]["conn_per_targetroute"][cpt]["keepalive"]:
```

Comment:
aakarsh would be proud.... 5 levels deep

Reply:
the data structure is 5 levels deep :wink: